### PR TITLE
Require JDK 22.0.0 to build and run Trino

### PR DESCRIPTION
**Pull request by wendigo**
## Additional context and related issues

* #20980

## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
(x) Release notes are required, with the following suggested text:

```markdown
# General
* Require JDK 22.0.0 to run Trino ({issue}`20980`)
```

### CODE REVIEW

File: docs/src/main/sphinx/installation/deployment.md [hunk 1]
```diff
@@ -35,9 +35,9 @@
 
 ### Java runtime environment
 
-Trino requires a 64-bit version of Java 21, with a minimum required version of 21.0.1.
-Earlier major versions such as Java 8, Java 11 or Java 17 do not work.
-Newer major versions such as Java 22 are not supported -- they may work, but are not tested.
+Trino requires a 64-bit version of Java 22, with a minimum required version of 22.0.0.
+Earlier major versions such as Java 8, Java 11, Java 17 or Java 21 do not work.
```
Comment (mosabua) [inline docs/src/main/sphinx/installation/deployment.md:39]:
```suggestion
Earlier versions such as Java 8, Java 11, Java 17, or Java 21 do not work.
```

File: docs/src/main/sphinx/installation/deployment.md [hunk 2]
```diff
@@ -35,9 +35,9 @@
...
+Newer major versions such as Java 23 are not supported -- they may work, but are not tested.
```
Comment (mosabua) [inline docs/src/main/sphinx/installation/deployment.md:40]:
```suggestion
Newer versions such as Java 23 are not supported -- they may work, but are not tested.
```

**Review by mosabua — APPROVED** [review]
Two mini changes for docs .. but otherwise great. Lets ship this.

**Comment by wendigo** [comment]
Requires released version of Airbase 155: https://github.com/airlift/airbase/pull/407 to support unnamed variables syntax

**Comment by mosabua** [comment]
As discussed - to be merged after Trino 446 is released.

**Pull request merged**